### DISCUSSION TITLE
I want to pass textVariant as a prop to my button component, what type should the prop be?

### DISCUSSION THREAD

**Discussion by AAfify1** [Shopify/restyle#199] [Q&A]
I am trying to pass a textVariant to my button prop to control the label text. However, when I define my prop as a string like so `type MyProps = {
  onPress: () => void;
  label: string;
  variant?: string;
  textVariant?: string;
};`

I get the typeScript error:

`Type 'string' is not assignable to type '"body" | "header" | "subHeader" | "welcomeButtons" | { phone?: "body" | "header" | "subHeader" | "welcomeButtons" | undefined; tablet?: "body" | "header" | "subHeader" | "welcomeButtons" | undefined; } | undefined'.` 

Is there a type export from the package that tracks the textVaraints and would be appropriate in this situation? Thanks!

**Comment by fabiocalliari-dev** [comment]
I have not tried it, but I'm pretty positive you can do this by exporting your Theme typings and declaring label as Theme['textVariant'].

**Comment by Charly6596** [comment]
`textVariant?: keyof Theme['textVariants'];` should do

For a built in type in restyle, you could also do the following:
```ts
type MyProps = VariantProps<Theme, "textVariants"> & {
  label: string;
}

const MyComponent = ({ label, variant }: MyProps) => {

}
```

But in your case, you already have a `variants` property
The only way I know to get the type of variants is using `VariantProps<Theme, "textVariants">['variant'];` so I'm interested if someone knows another way
